src/state/debugger.rs

```diff
@@ -47,17 +59,22 @@ impl Debugger {
         let contract = contract.copied().unwrap_or_default();
         let last_state = self.last_state.take();
 
+        let current = Breakpoint::raw(contract, pc);
```

Comment:
Maybe we move this alloc inside the `if self.single_stepping`? This way we prevent additional alloc for every cycle, regardless of the configuration of the debug

Reply:
it seems like this might actually be less allocations than before, since before we were making a breakpoint instance for each configured breakpoint in the set below using the identical pc settings.